app/bundles/ReportBundle/Controller/ReportController.php

```diff
@@ -630,8 +630,20 @@ public function viewAction($objectId, $reportPage = 1)
 
         // Setup dynamic filters
         $filterDefinitions = $model->getFilterList($entity->getSource());
-        $dynamicFilters    = $session->get('mautic.report.'.$objectId.'.filters', []);
-        $filterSettings    = [];
+        /** @var array $dynamicFilters */
+        $dynamicFilters = $session->get('mautic.report.'.$objectId.'.filters', []);
+        $filterSettings = [];
+
+        if (count($dynamicFilters) > 0 && count($entity->getFilters()) > 0) {
+            foreach ($entity->getFilters() as $fid => $filter) {
+                foreach ($dynamicFilters as $dfcol => &$dfval) {
```

Comment:
Do you have reason for using &? I would set it through an array, it is more readable for me.